### PR TITLE
Move sym_int and sym_float alongside SymInt / SymFloat in base torch package

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #91318
* __->__ #91317
* #91260

This PR moves the definitions for:
* `sym_int`
* `sym_ceil` (used only for `sym_int`)
* `sym_floor` (used only for `sym_int`)
* `sym_float`

from `torch/fx/experimental/symbolic_shapes.py` to `torch/__init__.py`, where `SymInt` and `SymFloat` are already defined.

This removes the need for several in-line imports, and enables proper JIT script gating for #91318. I'm very open to doing this in a better way!

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
Okey dokey

**Comment by jbschlosser** [comment]
FYI @ezyang to minimize introductions to the base `torch.` namespace, I tweaked this slightly so that only `sym_int` / `sym_float` are exposed there. This is still enough for #91318.

**Review by anijain2305 — APPROVED** [review]

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -931,10 +931,11 @@ def forward(self, a_1):
     return div""")
 
         r = str(make_fx(f, tracing_mode="symbolic", decomposition_table=decomposition_table)(torch.empty(4)).code).strip()
+        print('DEBUG:', r)
```
Comment (ezyang) [inline test/test_proxy_tensor.py]: don't forget to remove when done. Also, you do know about `EXPECTTEST_ACCEPT` right

Comment (jbschlosser) [reply test/test_proxy_tensor.py]: Oh nice :)

**Pull request closed without merging**